ts/main.ts

```diff
@@ -132,3 +138,270 @@ async function renderAllCards(): Promise<void> {
+  '#spell-details-description-div',
+) as HTMLDivElement;
+const $spellDetailsHigherLevelDiv = document.querySelector(
+  '#spell-details-higher-levels-div',
+) as HTMLDivElement;
+const $spellDetailsClasses = document.querySelector(
+  '#spell-details-classes',
+) as HTMLSpanElement;
+const $spellDetailsSubclasses = document.querySelector(
+  '#spell-details-subclasses',
+) as HTMLSpanElement;
+
+if (!$spellDetailsView) throw new Error('$spellDetailsView query failed');
+if (!$spellDetailsName) throw new Error('$spellDetailsName query failed');
+if (!$spellDetailsLevelSchool)
+  throw new Error('$spellDetailsLevelSchool query failed');
+if (!$spellDetailsCastTime)
+  throw new Error('$spellDetailsCastTime query failed');
+if (!$spellDetailsRange) throw new Error('$spellDetailsRange query failed');
+if (!$spellDetailsComponents)
+  throw new Error('$spellDetailsComponents query failed');
+if (!$spellDetailsDuration)
+  throw new Error('$spellDetailsDuration query failed');
+if (!$spellDetailsDescriptionDiv)
+  throw new Error('$spellDetailsDescriptionDiv query failed');
+if (!$spellDetailsHigherLevelDiv)
+  throw new Error('$spellDetailsHigherLevelDiv query failed');
+if (!$spellDetailsClasses) throw new Error('$spellDetailsClasses query failed');
+if (!$spellDetailsSubclasses)
+  throw new Error('$spellDetailsSubclasses query failed');
+
+interface SpellDetails {
+  name: string;
+  level: number;
+  school: {
+    name: string;
+  };
+  casting_time: string;
+  range: string;
+  components: string[];
+  duration: string;
+  desc: string[];
+  higher_level: string[] | null[];
+  classes: { name: string }[];
+  subclasses: { name: string }[];
+}
+
+function generateFullSubclassName(subclass: string): string {
+  switch (subclass) {
+    case 'Berserker':
+      return 'Barbarian Path of the Berserker';
+    case 'Champion':
+      return 'Fighter Champion Archetype';
+    case 'Devotion':
+      return 'Paladin Oath of Devotion';
+    case 'Draconic':
+      return 'Sorcerer Draconic Bloodline';
+    case 'Evocation':
+      return 'Wizard School of Evocation';
+    case 'Fiend':
+      return 'Warlock Fiend Patron';
+    case 'Hunter':
+      return 'Ranger Hunter Archetype';
+    case 'Land':
+      return 'Druid Circle of the Land';
+    case 'Life':
+      return 'Cleric Life Domain';
+    case 'Lore':
+      return 'Bard College of Lore';
+    case 'Open Hand':
+      return 'Monk Way of the Open Hand';
+    case 'Thief':
+      return 'Rogue Thief Archetype';
+    default:
+      return '';
+  }
+}
+
+let spellDetails: SpellDetails;
+
+$spellsListCardsDiv.addEventListener('click', async (event: Event) => {
+  try {
+    const $target = event.target as HTMLElement;
+
+    if (!$target.closest('div.card')) {
+      console.log('not a card');
+      return;
+    }
+
+    const $targetCard = $target.closest('div.card') as HTMLDivElement;
+    const cardSpellUrl = $targetCard.getAttribute('data-url');
```

Comment:
Since your line 250 is checking the closest div we could move line 255 above line 250. This way we get a variable, check the variable, and then get to use it in the future. Essentially querying for the same element only once.